src/lib.rs

```diff
@@ -54,3 +146,32 @@ pub use crate::stream::*;
 pub use crate::time::*;
 #[cfg(feature = "user")]
 pub use crate::user::*;
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn lol() {
+        assert!(broadcaster_id("literal"));
+        assert!(!broadcaster_id(String::from("string")));
+        assert!(broadcaster_id(UserIdRef::from_static("static ref")));
+        assert!(!broadcaster_id(UserId::new(String::from("owned"))));
+        assert!(!broadcaster_id(std::borrow::Cow::Owned(UserId::new(
+            String::from("cow owned")
+        ))));
+        assert!(broadcaster_id(std::borrow::Cow::Borrowed(
+            UserIdRef::from_static("cow borrowed")
+        )));
+    }
+    /// aa
+    pub fn broadcaster_id<'a>(broadcaster_id: impl IntoCow<'a, UserIdRef> + 'a) -> bool {
+        struct K<'a> {
+            id: std::borrow::Cow<'a, UserIdRef>,
+        }
+        let k = K {
+            id: broadcaster_id.to_cow(),
+        };
+        matches!(k.id, std::borrow::Cow::Borrowed(_))
+    }
```

Comment:
@Nerixyz I think this solves the issue described in https://github.com/twitch-rs/twitch_api/pull/280#issuecomment-1295168512

Reply:
Yes this should solve it 👍